Date: Fri, 17 Oct 2025 01:43:34 +0000
Subject: arm64: crypto: convert exported crypto symbol into pluggable
 interface for CONFIG_CRYPTO_AES_ARM64 crypto

Apply Crypto API wrappers to the exported crypto symbol in
CONFIG_CRYPTO_AES_ARM64-related crypto to convert them into pluggable
interface.

Signed-off-by: Jay Wang <wanjay@amazon.com>

diff --git a/arch/arm64/crypto/Makefile b/arch/arm64/crypto/Makefile
--- a/arch/arm64/crypto/Makefile
+++ b/arch/arm64/crypto/Makefile
@@ -50,7 +50,7 @@ aes-neon-blk-y := aes-glue-neon.o aes-neon.o
 crypto-objs-$(CONFIG_CRYPTO_NHPOLY1305_NEON) += nhpoly1305-neon.o
 nhpoly1305-neon-y := nh-neon-core.o nhpoly1305-neon-glue.o
 
-obj-$(CONFIG_CRYPTO_AES_ARM64) += aes-arm64.o
+crypto-objs-$(CONFIG_CRYPTO_AES_ARM64) += aes-arm64.o
 aes-arm64-y := aes-cipher-core.o aes-cipher-glue.o
 
 obj-$(CONFIG_CRYPTO_AES_ARM64_BS) += aes-neon-bs.o
diff --git a/arch/arm64/crypto/aes-cipher-glue.c b/arch/arm64/crypto/aes-cipher-glue.c
--- a/arch/arm64/crypto/aes-cipher-glue.c
+++ b/arch/arm64/crypto/aes-cipher-glue.c
@@ -54,8 +54,8 @@ static void __exit aes_fini(void)
 	crypto_unregister_alg(&aes_alg);
 }
 
-module_init(aes_init);
-module_exit(aes_fini);
+crypto_module_init(aes_init);
+crypto_module_exit(aes_fini);
 
 MODULE_DESCRIPTION("Scalar AES cipher for arm64");
 MODULE_AUTHOR("Ard Biesheuvel <ard.biesheuvel@linaro.org>");
